scripts/qupath_export_label_image.groovy

```diff
@@ -0,0 +1,45 @@
+import qupath.lib.regions.*
+import ij.*
+import java.awt.*
+import java.awt.image.BufferedImage
+import javax.imageio.ImageIO
+
+def server = getCurrentImageData().getServer()
+int w = server.getWidth()
+int h = server.getHeight()
+
+int[][] labelArray = new int[h][w]
+
+int index = 1
+for (detection in getDetectionObjects()) {
+    roi = detection.getROI()
+    
+    for (int y = (int)roi.getBoundsY(); y <= (int)(roi.getBoundsY() + roi.getBoundsHeight()); y++){
+        for (int x = (int)roi.getBoundsX(); x <= (int)(roi.getBoundsX() + roi.getBoundsWidth()); x++) {
+            if (roi.contains(x, y)) {
+                labelArray[y][x] = index
+            }
+        }
+    }    
+    index++
+}
+
+def labelImage = new BufferedImage(w, h, BufferedImage.TYPE_USHORT_GRAY)
+
+def g2d = labelImage.createGraphics()
+
+for (int y = 0; y < h; y++) {
+    for (int x = 0; x < w; x++) {
+        labelValue = labelArray[y][x]
+        labelImage.getRaster().setPixel(x, y, labelValue)
+    }
+}
+
+g2d.dispose()
+
+new ImagePlus('Labels', labelImage).show()
+
+def name = getProjectEntry().getImageName()
+def path = buildFilePath(PROJECT_BASE_DIR)
+def fileoutput = new File(path, 'labels.png')
```

Comment:
This saves the file as 'labels.png' in the top level of the project directory, but if I have multiple images in my project, then it'll overwrite each one if I run it on each.

Reply:
I can append "labels" to the end of the name as supplied by QuPath, but it is formatted "[original file name].tiff - [original file name].tiff SIMS - labels.png". That seems to be a limitation of how QuPath builds an image name and I don't immediately see a way around it.

Reply:
I was going to suggest some string manipulation such as 
```
def imageName = getProjectEntry().getImageName()
def originalName = imageName.split('.tiff - ')[0]
```
but I believe the `getOriginalImageName` method is actually what you want:
```
def imageName = getProjectEntry().getOriginalImageName()
```